From baaa744b53338bec3fd7ac1881f35811fa757231 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:11 -0400
Subject: [usb] xhci: remove unnecessary check in xhci_free_stream_info()

Message-id: <1410887463-20674-180-git-send-email-dzickus@redhat.com>
Patchwork-id: 95552
O-Subject: [RHEL7 PATCH 179/591] xhci: remove unnecessary check in xhci_free_stream_info()
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 0d3703be3a671ca225a796f4ae6e904dd83c2e07
Author: Xenia Ragiadakou <burzalodowa@gmail.com>
Date:   Mon Aug 26 23:29:48 2013 +0300

    xhci: remove unnecessary check in xhci_free_stream_info()

    This patch removes the unneccessary check 'if (stream_info)' because
    there is already a check few lines above which ensures that stream_info
    is not NULL.

    Signed-off-by: Xenia Ragiadakou <burzalodowa@gmail.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci-mem.c b/drivers/usb/host/xhci-mem.c
index 90709cf..1445e08 100644
--- a/drivers/usb/host/xhci-mem.c
+++ b/drivers/usb/host/xhci-mem.c
@@ -722,8 +722,7 @@ void xhci_free_stream_info(struct xhci_hcd *xhci,
     stream_info->stream_ctx_array,
     stream_info->ctx_array_dma);
 
- if (stream_info)
-  kfree(stream_info->stream_rings);
+ kfree(stream_info->stream_rings);
  kfree(stream_info);
 }
 
-- 
1.7.1